File: backend/app/schemas/room.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RoomBase(BaseModel):
    name: str
    room_type: str
    capacity: int
    is_active: bool = True


class RoomCreate(BaseModel):
    name: str
    room_type: str
    capacity: int


class RoomRead(RoomBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class RoomUpdate(BaseModel):
    name: str | None = None
    room_type: str | None = None
    capacity: int | None = None
    is_active: bool | None = None

File: frontend/src/types/room.ts
// Mirrors backend/app/schemas/room.py in full. capacity is `| null` --
// Task 14 made Room.capacity optional on the backend, so the frontend type
// must allow null too, not just make the field itself optional.

export interface Room {
  id: number;
  name: string;
  room_type: string;
  capacity: number | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface RoomCreate {
  name: string;
  room_type: string;
  capacity?: number | null;
}

export interface RoomUpdate {
  name?: string;
  room_type?: string;
  capacity?: number | null;
  is_active?: boolean;
}
